### PR TITLE
Parser: Use `as` syntax in namespace patterns

### PR DESCRIPTION
**Pull request by feds01**

### CODE REVIEW

File: docs/src/features/patterns.md [hunk 1]
```diff
@@ -259,21 +259,21 @@ or_pattern = ( single_pattern "|" )+ single_pattern
 
 binding_pattern = identifier
 
-tuple_pattern_member = identifier | ( identifier "=" single_pattern )
+tuple_pattern_member = identifier | ( identifier "=" identifier )
```
Comment (kontheocharis) [inline docs/src/features/patterns.md:262]: Why is this changed?

Comment (feds01) [reply docs/src/features/patterns.md:262]: I thought that it doesn't make sense to attach anything but a bind to the range pattern
